### PR TITLE
Update to pytest 3.0.1

### PR DESCRIPTION
**Pull request by dwaynebailey**
Changes include:
* Update to pytest 3.0.1 and pytest-django 3.0.0
* Updates to supporting test infrastructure: pytest-cov and pytest-caplog
* Fixes to get tests running
* Cleanups for deprecated features

### CODE REVIEW

**Comment by dwaynebailey** [comment]
Would appreciate thoughts on:

* 810c3e2 - I'm not sure if this is something we want sitewide or just for tests, I don't know logging module and logging config well enough.
* 69a033a vs 	945281e - I'm not sure which one is actually needed.  Likely they can just be squashed and 945281e is what we want.  But would love feedback.

**Comment by unho** [comment]
Nit: Fix the typos in the commit messages before landing.

**Comment by unho** [comment]
Looks good, except for those commits asked for input which I can't tell.

**Comment by unho** [comment]
lgtm

**Comment by phlax** [comment]
lgtm

File: pootle/settings/91-travis.conf [hunk 1]
```diff
@@ -23,6 +23,9 @@ if os.environ.get("TRAVIS"):
             'HOST': '',
             'PORT': '',
             'ATOMIC_REQUESTS': True,
+            'TEST': {
+                'NAME': 'test_pootle',
+            }
```
Comment (phlax) [inline pootle/settings/91-travis.conf]: @dwaynebailey i think we need to update this in the default dbs dict, and probs in the `pootle init` command too

File: pytest_pootle/fixtures/site.py [hunk 2]
```diff
@@ -26,11 +26,19 @@ def setup_db_if_needed(request):
...
 @pytest.fixture(scope='session')
-def post_db_setup(translations_directory, _django_db_setup,
-                  _django_cursor_wrapper, request):
+def tests_use_db(request):
+    return bool(
+        [item for item in request.node.items
+         if item.get_marker('django_db')])
+
+
+@pytest.fixture(autouse=True, scope='session')
```
Comment (phlax) [inline pytest_pootle/fixtures/site.py]: this one shouldnt be autouse

**Pull request merged**